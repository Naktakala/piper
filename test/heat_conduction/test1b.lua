nodes = {}
N = 10
xmin = 0.0
L = 1.0
nodes = {}

dx = L / N
for i = 0, N do
  nodes[i + 1] = xmin + dx * i
end
meshgen1 = chi_mesh.OrthogonalMeshGenerator.Create({ node_sets = { nodes } })
chi_mesh.MeshGenerator.Execute(meshgen1)

hcsystem = hcm.HeatConductionSystem.Create({
  kernels = {
    hcm.ThermalConductionKernel.Create({ k = 16.0 }),
    chi_math.SinkSourceFEMKernel.Create({ value = 100.0e2 })
  },
  bcs = {
    chi_math.FEMDirichletBC.Create({ boundaries = { "ZMAX" }, bc_value = 1.0 }),
    chi_math.FEMDirichletBC.Create({ boundaries = { "ZMIN" }, bc_value = 1.0 })
  }
})

phys1 = hcm.HCSteadyExecutor.Create({
  conduction_system = hcsystem,
  solver_params =
  {
    --l_max_its = 10
    l_rel_tol = 1.0e-5,
    nl_method = "JFNK"
  }
})

chi.AggregateNodalValuePostProcessor.Create
({
  name = "maxval",
  field_function = "T",
  operation = "max",
  print_on = {"ProgramExecuted"}
})

chi.PostProcessorPrinterSetOptions
(
  {
    print_scalar_time_history = false
  }
)

chiSolverInitialize(phys1)
chiSolverExecute(phys1)

--chiExportMultiFieldFunctionToVTK({ "T" }, "test1")