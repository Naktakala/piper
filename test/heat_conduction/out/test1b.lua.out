mpiexec -np 2 /Users/vermjc/Repos/ChiTech/piper/bin/piper test1b.lua --suppress_color --supress_beg_end_timelog master_export=false 
[0]  Parsing argument 1 test1b.lua
[0m[0]  Parsing argument 2 --suppress_color
[0m[0]  Parsing argument 3 --supress_beg_end_timelog
[0]  Parsing argument 4 master_export=false
[0]  ChiTech number of arguments supplied: 4
[0]  Computing cell-centroids.
[0]  Done computing cell-centroids.
[0]  Checking cell-center-to-face orientations
[0]  Done checking cell-center-to-face orientations
[0]  00:00:00 Establishing cell connectivity.
[0]  00:00:00 Vertex cell subscriptions complete.
[0]  00:00:00 Surpassing cell 1 of 10 (10%)
[0]  00:00:00 Surpassing cell 2 of 10 (20%)
[0]  00:00:00 Surpassing cell 4 of 10 (30%)
[0]  00:00:00 Surpassing cell 5 of 10 (40%)
[0]  00:00:00 Surpassing cell 6 of 10 (50%)
[0]  00:00:00 Surpassing cell 7 of 10 (60%)
[0]  00:00:00 Surpassing cell 8 of 10 (70%)
[0]  00:00:00 Surpassing cell 9 of 10 (80%)
[0]  00:00:00 Surpassing cell 10 of 10 (90%)
[0]  00:00:00 Establishing cell boundary connectivity.
[0]  00:00:00 Done establishing cell connectivity.
[0]  MeshGenerator: Cells created = 10
[0]  
[0]  Executing solver "phys1"
[0]  NonLinearSolver non-linear iteration    0 - Residual 3.1130692e+03
[0]  NonLinearSolver iteration    0 - Residual 3.1130692e+03
[0]  NonLinearSolver iteration    1 - Residual 2.7503252e+03
[0]  NonLinearSolver iteration    2 - Residual 2.3582437e+03
[0]  NonLinearSolver iteration    3 - Residual 1.8871277e+03
[0]  NonLinearSolver iteration    4 - Residual 1.2613800e+03
[0]  NonLinearSolver iteration    5 - Residual 1.8575776e-04
[0]  NonLinearSolver non-linear iteration    1 - Residual 1.0973067e-03
[0]  NonLinearSolver iteration    0 - Residual 1.0973067e-03
[0]  NonLinearSolver iteration    1 - Residual 4.5337971e-04
[0]  NonLinearSolver iteration    2 - Residual 2.3011660e-04
[0]  NonLinearSolver iteration    3 - Residual 1.6758760e-04
[0]  NonLinearSolver iteration    4 - Residual 6.5879472e-05
[0]  NonLinearSolver iteration    5 - Residual 3.5294062e-05
[0]  NonLinearSolver iteration    6 - Residual 2.6712979e-05
[0]  NonLinearSolver iteration    7 - Residual 2.2406782e-05
[0]  NonLinearSolver iteration    8 - Residual 1.9633193e-05
[0]  NonLinearSolver iteration    9 - Residual 5.6476444e-19
[0]  NonLinearSolver non-linear iteration    2 - Residual 2.0827950e-10
[0]            Converged CONVERGED_FNORM_ABS
[0]  
[0]  
[0]  SCALAR post-processors latest values at event "ProgramExecuted"
[0]  *----------------------*-----------------*
[0]  | Post-Processor Name  | Value           |
[0]  *----------------------*-----------------*
[0]  | maxval(latest)       |       79.125000 |
[0]  *----------------------*-----------------*
[0]  


